src/org/labkey/targetedms/outliers/OutlierGenerator.java

```diff
@@ -64,14 +64,31 @@ private String getEachSeriesTypePlotDataSql(int seriesIndex, int id, String sche
             StringBuilder filterClause = new StringBuilder("SampleFileId.ReplicateId in (");
             var intersect = "";
             var selectSql = "(SELECT ReplicateId FROM targetedms.ReplicateAnnotation WHERE ";
+            var escapeQuote = "\'";
             for (AnnotationGroup annotation : annotationGroups)
             {
-                filterClause.append(intersect).append(selectSql).append(" Name='").append(annotation.getName()).append("' AND ( ");
-                var or = "";
-                for (String value : annotation.getValues())
+                filterClause.append(intersect)
+                        .append(selectSql)
+                        .append(" Name=")
+                        .append(escapeQuote)
+                        .append(annotation.getName())
+                        .append(escapeQuote);
+
+                var annotationValues = annotation.getValues();
+                if (!annotationValues.isEmpty())
                 {
-                    filterClause.append(or).append("Value='").append(value).append("'");
-                    or = " OR ";
+                    filterClause.append(" AND  Value IN (")
+                            .append(escapeQuote)
+                            .append(annotationValues.get(0))
```

Comment:
Suggest using StringUtils.join() to build up the quote-escaped, comma-separated list

Reply:
I guess you meant this: 
`var quoteEscapedVals = annotationValues.stream().map(s -> s.replace("'", "''")).collect(Collectors.toList());
filterClause.append(" AND  Value IN (").append(StringUtils.join(quoteEscapedVals, ","));`

still a for loop would be required to build quotes around the strings?

Reply:
done